Userland/Libraries/LibCore/EventLoop.cpp

```diff
@@ -422,9 +423,11 @@ void EventLoop::pump(WaitMode mode)
                 new_event_queue.unchecked_append(move(events[i]));
             new_event_queue.extend(move(m_queued_events));
             m_queued_events = move(new_event_queue);
-            return;
+            break;
         }
     }
+
+    return processed_events;
```

Comment:
:thinking: how do you plan to use this exactly?
This will either tell you that it did process something, or it will block for a while, then tell you that it processed something...
Maybe a `spin_until_idle()` might be better?

Reply:
Exactly like this:
https://github.com/SerenityPorts/SDL/pull/26

I would need something like `spin_until_idle_or_timeout()` which would be a bit too specific for my taste. The only use case I have so far is for SDL2, and the way I implemented it there is the same as how the X11 or Windows backends deal with it. But I'm open to suggestions!

Reply:
Oh I see, that uses `PollForEvents`
The return value seem entirely useless for the default mode, but whatever, it's merged now